src/Microsoft.Azure.SignalR.Common/ServiceConnections/MultiEndpointServiceConnectionContainer.cs

```diff
@@ -115,11 +116,14 @@ public Task ConnectionInitializedTask
 
         public Task StartAsync()
         {
-            return Task.WhenAll(_routerEndpoints.endpoints.Select(s =>
-            {
-                Log.StartingConnection(_logger, s.Endpoint);
-                return s.ConnectionContainer.StartAsync();
-            }));
+            //ensure started only once
+            return _started == 1 || Interlocked.CompareExchange(ref _started, 1, 0) == 1
```

Comment:
`Interlocked.CompareExchange` is enough. BTW  `_started ==1` does not necessarily get the latest modified value, you need `Interlocked.Read`

Reply:
`_started==1` is just intended to avoid using `Interlocked` in most cases. Its effect should be similar as the previous `TryGetOrCreate`. So the latest modified value is not necessary.

Reply:
Ah I c